internal/agent/worker.go

```diff
@@ -428,7 +429,12 @@ func (w *Worker) deleteProbe(ctx context.Context, shutdownChan chan struct{}) er
 		}
 		err := w.probeManager.DeleteProbeK8sResource(probe)
 		if err != nil {
-			return fmt.Errorf("failed to delete CR for probe %s: %w", probe.ID, err)
+			// Check if the error is due to missing CRDs - in this case, we should still proceed with API deletion
```

Comment:
Let's treat "no compatible Probe CRDs" as a warning, not a fatal error:
```suggestion
      // If the CRD doesn't exist or CR is already gone, log warning and continue with API cleanup
      logger.Warnf("failed to delete CR for probe %s (may already be deleted): %v", probe.ID, err)
      // Don't return error - continue to API deletion
```